cloud/pkg/controller/manager/location.go

```diff
@@ -10,12 +10,14 @@ import (
 
 // LocationCache cache the map of node, pod, configmap, secret
 type LocationCache struct {
-	//edgeNodes is a list of valid edge nodes
-	edgeNodes []string
+	// EdgeNodes is a list of valid edge nodes
+	EdgeNodes []string
```

Comment:
Using sync.Map(key: nodename; value: state) instead of this "EdgeNodes" string slice here would be much better. Reasons:
1. Using sync.Map makes the time complexity of function UpdateEdgeNode and IsEdgeNode O(1), while using string slice with for loop makes it O(n).
2. Later we could be checking node state whether it's "ready" or still "not ready" by directly comparing "value" of sync.Map and doing the node state updating.

Reply:
yess, I have added this TODO https://github.com/kubeedge/kubeedge/pull/674/files#diff-55277caec6421504e2848e75c588c384R198 ... I will make this change in this PR itself

Reply:
I have used map[string]string with lock instead of sync.Map as sync.Map doesn't allow to range(sync.Map)  and we need to range over all nodes to send services and endpoints to all nodes.

Reply:
sync.Map allow range, continue when f return true, break when f return false
```
// Range calls f sequentially for each key and value present in the map.
// If f returns false, range stops the iteration.
//
// Range does not necessarily correspond to any consistent snapshot of the Map's
// contents: no key will be visited more than once, but if the value for any key
// is stored or deleted concurrently, Range may reflect any mapping for that key
// from any point during the Range call.
//
// Range may be O(N) with the number of elements in the map even if f returns
// false after a constant number of calls.
func (m *Map) Range(f func(key, value interface{}) bool) {
}
```

Reply:
Thanks Yushun. Used syncMap